File: java/json-transform/src/main/java/ai/bizone/jsontransform/functions/TransformerFunctionTransform.java
package ai.bizone.jsontransform.functions;

import ai.bizone.jsontransform.functions.common.*;
import ai.bizone.jsontransform.functions.common.*;

import java.util.Map;

public class TransformerFunctionTransform extends TransformerFunction {
    public TransformerFunctionTransform() {
        super(FunctionDescription.of(
            Map.of(
            "to", ArgumentType.of(ArgType.Any).position(0)
            )
        ));
    }
    @Override
    public Object apply(FunctionContext context) {
        var input = context.getJsonElement(null);
        var to = context.getJsonElement("to", false);
        return context.transformItem(to, input);
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionTransform.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import { ArgType } from "./common/ArgType";

class TransformerFunctionTransform extends TransformerFunction {
  constructor() {
    super({
      argsSets: [[{ name: "to", type: ArgType.Any }]],
    });
  }

  override async apply(context: FunctionContext): Promise<any> {
    const input = await context.getJsonElement(null);
    const to = await context.getJsonElement("to", false);
    return context.transformItem(to, input);
  }
}

export default TransformerFunctionTransform;
